service/matching/taskQueueManager.go

```diff
@@ -517,11 +519,17 @@ func (c *taskQueueManagerImpl) GetUserData(ctx context.Context) (*persistencespb
 	if c.managesSpecificVersionSet() {
 		return nil, nil, errNoUserDataOnVersionedTQM
 	}
+	if !c.config.LoadUserData() {
+		return nil, nil, nil
```

Comment:
Did you consider having this function return a sentinel error in this case and having the other "disable versioning" stuff look at that error instead of separately evaluating `config.LoadUserData()`? At first glance, I think that would be cleaner since the config is only being used in one place

Reply:
I like that this config is fairly contained in the code and doesn't leak all over.
Remember that this is a temporary flag (or so we hope) and I want to be able to easily remove it later.

Reply:
I tried this refactoring and I like it so far.. I'll send another PR later for discussion